src/mbgl/layout/symbol_instance.hpp

```diff
@@ -21,23 +21,39 @@ struct ShapedTextOrientations {
     bool singleLine = false;
 };
 
+struct SymbolInstanceSharedData {
+    SymbolInstanceSharedData(GeometryCoordinates line,
+                            const ShapedTextOrientations& shapedTextOrientations,
+                            optional<PositionedIcon> shapedIcon,
+                            const style::SymbolLayoutProperties::Evaluated& layout,
+                            const float layoutTextSize,
+                            const style::SymbolPlacementType textPlacement,
+                            const std::array<float, 2>& textOffset,
+                            const GlyphPositions& positions);
+    bool empty() const;
+    GeometryCoordinates line;
```

Comment:
Which of these std::vectors: line, rightJustifiedGlyphQuads ... verticalGlyphQuads are shared to multiple places?

I'm just asking if it is necessary to have them all here in struct (and manage their lifecycle together via releaseSharedData ). If some of them are not shared, could we get the same effect if using std::move and swap(emptyVector) to control release?

Reply:
they are all in one struct `SymbolInstanceSharedData` and released via `SymbolInstance::releaseSharedData()`